packages/calypso-e2e/src/lib/components/comments-likes-component.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Internal dependencies
+ */
+import { BaseContainer } from '../base-container';
+
+/**
+ * Type dependencies
+ */
+import { Page, ElementHandle } from 'playwright';
+
+const selectors = {
+	topLevelComments: '.comment-list .depth-1:not(.children)',
+	likeButton: '.comment-like-link',
+	notLiked: '.comment-not-liked',
+	liked: '.comment-liked',
+};
+
+/**
+ * Component representing the like component of a comment.
+ *
+ * @augments {BaseContainer}
+ */
+export class CommentsLikesComponent extends BaseContainer {
+	/**
+	 * Constructs an instance of the object.
+	 *
+	 * @param {Page} page Instance of the page on which the component resides.
+	 */
+	constructor( page: Page ) {
+		super( page, selectors.topLevelComments );
+	}
+
+	/**
+	 * Overrides the parent method for post-initialization steps.
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async _postInit(): Promise< void > {
+		await this.page.waitForLoadState( 'networkidle' );
+	}
+
+	/**
+	 * Clicks the Like button and toggles the state.
+	 *
+	 * This function will also confirm that click action on the Like button
+	 * had the intended effect.
+	 *
+	 * @param {number} commentNum The nth comment to click like on. Defaults to 1.
+	 * @returns {Promise<void>} No return value.
+	 * @throws {Error} If no comments exist.
+	 */
+	async clickLikeComment( commentNum = 1 ): Promise< void > {
+		await this.page.waitForSelector( selectors.topLevelComments );
+
+		// For now, only works with top level comments since child comments introduce
+		// a whole new layer of complexity.
+		const topLevelComments = await this.page.$$( selectors.topLevelComments );
```

Comment:
Would this work when replaced with the following?
```suggestion
		const topLevelComments = await this.page.waitForSelector( selectors.topLevelComments );
```

Reply:
Here, the `$$` selector returns `ElementHandle[]` which is what I want to do here.

`page.waitForSelector` matches on only one element and would return the first one that meets the criteria (visible, etc). Since I am trying to retrieve the entire list of top level comments for further processing, the `$$` call is more appropriate here.

The reason why I excluded child comments has to do with a few reasons:
- inconsistent selector - there isn't a good way to select `just` the child comments that I've found so far.
- what comment number to assign child comments - if the first comment has a reply, would that be considered second? What if there's further replies along? Would these comments bump rest of the comments down on the index?

On top of that, there isn't a test that tests replies/child comments at this moment in time. I figured I'd address the complexities of abstracting parent/child comments down the road when I write new tests.

Reply:
Ah, TIL that `page.$$` is basically a `document.querySelectorAll` 👍 

Regarding the post selector - maybe we could base the selector off the comment's text instead of number? We're posting the comment we're liking, so it should be all we need to know to make a selector:

```js
function like( comment ) {
  await this.page.click( `a.comment-like-link:below(p:text(${ comment }))` );
  // ...and await here for the comment to become liked
}
```

Reply:
That's an idea, yeah. I can add a code path for liking a comment based on the comment content, not by the nth order by which it appears on page.

I still want to keep the code path for liking based on the nth number though, because I can see it being potentially useful for other tests.

Reply:
I've implemented your suggested changes in the latest push (support selection of comment by the comment body) while keeping the `nth` based selector path as well.